### PR TITLE
Split ContactsCollection mailet

### LINKED ISSUE #1257: Split ContactsCollection
### Why?

Because it does both indexing of the contact locally and forwards it to OpenPaaS

However with https://github.com/linagora/tmail-backend/issues/1163  OpenPaaS is the source of truth and would anyway forward back collected contacts to James if any

CF

image

We thus do not want given that feature to index contacts locally.

(We do not want contact auto-collection to override openpaas Data)

### How

Drop `ContactsCollection`

Use directly James' `ContactExtractor` for interop with OpenPaaS instread.

For standalone James write a `IndexContact` mailet indexing the contact locally.

### PR DESCRIPTION
**Pull request by HoussemNasri**
Closes #1257

## TODO

- [x] Add more integration tests
- [x] In James-project: fix `com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannot construct instance of org.apache.james.transport.mailets.ContactExtractor$ExtractedContacts`
- [x] Update documentation for the mailet
- [x] Delete `ContactsCollection` leftovers (code, doc, tests, etc...)
     - [x] Remove the replace_openpaas_amqp_uri hack from `MemoryCalendarEventReplyWithAMQPWorkflowTest` and use `OpenPaasAmqpForwardAttribute`.

### CODE REVIEW

File: tmail-backend/mailets/src/main/scala/com/linagora/tmail/mailets/IndexContacts.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+            MailAddress userMailAddress = new MailAddress(extractedContacts.getUserEmail());
+            return Flux.fromIterable(toContactFields(extractedContacts.getEmails()))
+                .flatMap(contact ->
+                    eventBus.dispatch(
+                        new TmailContactUserAddedEvent(
+                            Event.EventId.random(),
+                            Username.fromMailAddress(userMailAddress),
+                            contact),
+                        NO_REGISTRATION_KEYS)
+                ).collectList()
```
Comment (chibenwa) [inline tmail-backend/mailets/src/main/scala/com/linagora/tmail/mailets/IndexContacts.java:121]:
```suggestion
                        NO_REGISTRATION_KEYS))
                    .collectList()
```

File: tmail-backend/mailets/src/main/scala/com/linagora/tmail/mailets/IndexContacts.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+    @Override
+    public void service(Mail mail) throws MessagingException {
+        Mono.justOrEmpty(mail.getAttribute(attributeName))
+            .map(attribute -> attribute.getValue().asAttributeValueOf(String.class))
+            .flatMap(Mono::justOrEmpty)
+            .map(AttributeValue::value)
+            .map(Throwing.function(this::parseUserContacts))
+            .flatMap(Throwing.function(this::dispatchContactsAddedEvent))
+            .onErrorMap(e -> new MailetException("Failed to index contacts", new Exception(e)))
```
Comment (chibenwa) [inline tmail-backend/mailets/src/main/scala/com/linagora/tmail/mailets/IndexContacts.java:90]: We are not on a critical feature, so we should WARN on error and avoid propagating the exception

File: tmail-backend/mailets/src/main/scala/com/linagora/tmail/mailets/IndexContacts.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
+package com.linagora.tmail.mailets;
```
Comment (chibenwa) [inline tmail-backend/mailets/src/main/scala/com/linagora/tmail/mailets/IndexContacts.java:1]: Shall we remove the old class?

Comment (HoussemNasri) [reply tmail-backend/mailets/src/main/scala/com/linagora/tmail/mailets/IndexContacts.java:1]: It's removed now.

**Review by chibenwa — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

File: docs/modules/ROOT/pages/tmail-backend/configure/index.adoc [hunk 4]
```diff
@@ -249,11 +249,11 @@ The mailet configuration requires the following parameters:
 * `exchange_type`: type of the exchange. Can be "direct", "fanout", "topic", "headers". Optional, defaults to "direct".
 
 === Example
-This configuration forwards contacts collected by the `ContactsCollection` mailet to OpenPaas through the AMQP protocol.
+This configuration forwards contacts extracted by the `ContactExtractor` mailet to OpenPaas through the AMQP protocol.
 
 [source,xml]
 ----
-<mailet match="SenderIsLocal" class="com.linagora.tmail.mailets.ContactsCollection">
+<mailet match="SenderIsLocal" class="ContactExtractor">
```
[the remark this answers is not in the document]
Comment (vttranlina) [reply docs/modules/ROOT/pages/tmail-backend/configure/index.adoc:256]: maybe mobile team mount outdated configuration file to container

**Pull request merged**